Application/Services/AuthService.cs

```diff
@@ -0,0 +1,111 @@
+﻿using Application.DTOs;
+using Application.Interfaces;
+using AutoMapper;
+using Domain.Interfaces;
+using Domain.Model;
+
+namespace Application.Services;
+
+/// <summary>
+/// Represents authorization operations.
+/// </summary>
+public class AuthService : IAuthService
+{
+    private readonly IUserRepository userRepository;
+    private readonly IPasswordService passwordService;
+    private readonly IJWTTokenService jwtTokenService;
+    private readonly IMapper mapper;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AuthService"/> class.
+    /// </summary>
+    /// <param name="userRepository">User repository that allows database operations on user table.</param>
+    /// <param name="passwordService">Represents repository that allows password operations.</param>
+    /// <param name="jwtTokenService">Represents repository that allows generating JWT token for the user.</param>
+    /// <param name="mapper">Used to map objects between different classes.</param>
+    public AuthService(
+        IUserRepository userRepository,
+        IPasswordService passwordService,
+        IJWTTokenService jwtTokenService,
+        IMapper mapper)
+    {
+        this.userRepository = userRepository;
+        this.passwordService = passwordService;
+        this.jwtTokenService = jwtTokenService;
+        this.mapper = mapper;
+    }
+
+    /// <summary>
+    /// Method that logs in the user.
+    /// </summary>
+    /// <param name="userLoginDto">User's id, email, nickname and JWT token.</param>
+    /// <returns>User's data.</returns>
+    public async Task<LoginUserReturnDto> LoginUserAsync(LoginUserDto userLoginDto)
+    {
+        var user = await this.userRepository.GetUserByEmailAsync(userLoginDto.Email);
+
+        if (user != null && this.passwordService.VerifyPassword(user.PasswordHash, userLoginDto.Password))
```

Comment:
Po pierwsze: nazwa klasy `LoginUserDto`, a zmiennej `userLoginDto` - proponuję nie zamieniać miejscami wyrazów.
Po drugie: Komentarz sugeruje, że `LoginUserDto` zawiera nickname i JWT token, co nie jest prawdą.

```suggestion
    /// <summary>
    /// Authenticates the user by verifying credentials and returns login data.
    /// </summary>
    /// <param name="loginUserDto">The user's login credentials.</param>
    /// <returns>The user's login data or <see langword="null"/> if authentication fails.</returns>
    public async Task<LoginUserReturnDto> LoginUserAsync(LoginUserDto loginUserDto)
    {
        var user = await this.userRepository.GetUserByEmailAsync(loginUserDto.Email);

        if (user != null && this.passwordService.VerifyPassword(user.PasswordHash, loginUserDto.Password))
```